specs/agents/tracing-instrumentation-messaging.md

```diff
@@ -0,0 +1,105 @@
+
+### Context fields
+
+- **`context.message.queue.name`**: optional for `messaging` spans and transactions. Indexed as keyword. Wherever the broker terminology 
+uses "topic", this field will contain the topic name. In RabbitMQ, whenever the queue name is not available, use exchange name instead.
+- **`context.message.age.ms`**: optional for message/record receivers only (transactions or spans). 
+A numeric field indicating the message's age in milliseconds. Relevant for transactions and 
+`receive` spans that receive valid messages. There is no accurate definition as to how this is calculated. If the messaging framework 
+provides a timestamp for the message- agents may use it (subtract the message/record timestamp from the read timestamp). 
+If a timestamp is not available- agents should omit this field or find an alternative and document it in this spec. For example, the 
+sending agent can add a timestamp to the message's metadata to be retrieved by the receiving agent.  
+Clock skews between agents are ignored, unless the calculated age (receive-timestamp minus send-timestamp) is negative, in which case the 
+agent should report 0 for this field.
+- **`context.message.routing_key`**: optional. Use only where relevant. Currently only RabbitMQ.
+
+#### Transaction context fields
+
+- **`context.message.body`**: similar to HTTP requests' `context.request.body`- only fill in messaging-related **transactions** (ie 
+incoming messages creating a transaction) and not for outgoing messaging spans. 
+   - Capture only when `ELASTIC_APM_CAPTURE_BODY` config is set to `true`.
+   - Only capture UTF-8 encoded message bodies.
+   - Limit size to 10000 characters. If longer than this size, trim to 9999 and append with ellipsis
+- **`context.message.headers`**: similar to HTTP requests' `context.request.headers`- only fill in messaging-related **transactions**.
+   - Capture only when `ELASTIC_APM_CAPTURE_HEADERS` config is set to `true`.
+   - Sanitize headers with keys configured through `ELASTIC_APM_SANITIZE_FIELD_NAMES`.
+   - Intake: key-value pairs, same like `context.request.headers`.
+- **`context.service.framework.name`**: same as `span.subtype`, but not in lowercase, e.g. `Kafka`, `RabbitMQ`
+
+#### Span context fields
+
+- **`context.destination.address`**: optional. Not available in some cases. Only set if the actual connection is available.
+- **`context.destination.port`**: optional. Not available in some cases. Only set if the actual connection is available.
+- **`context.destination.service.name`**: mandatory. Use the framework name in lowercase, e.g. `kafka`, `rabbitmq`.
+- **`context.destination.service.resource`**: mandatory. Use the framework name in lowercase. Wherever the queue/topic/exchange name is
+ available, append it with a leading `/`, for example: `kafka/myTopic`, `rabbitmq/myExchange`, `rabbitmq`.
```

Comment:
we can remove `rabbitmq` as we always have exchange name here now, either `rabbitmq/<default>` `rabbitmq/<unknown>` or `rabbitmq/exchange`.

Reply:
So it is not according to the spec 🙂 .
In JMS we use `jms` when the destination is not known and in Kafka we use `kafka` for polls. We don't use `<unknown>` in neither.

Reply:
It used to be that initially:
- `rabbitmq` when exchange name is not known
- `rabbitmq/exchange` when it's known

From what I remember on our last discussions on RabbitMQ PR, we decided to always include the normalized exchange name for consistency, otherwise the simple `rabbitmq` would be confusing, especially in map (should it target the whole rabbitmq resource or only a sub-division of it)
- `rabbitmq/<unknown>` when exchange is not known
- `rabbitmq/<default>` when using default exchange
- `rabbitmq/exchange` when exchange is known and is non-default

With that, weither the exchange is known or not, every rabbitmq resource node in the map will be at the same level, and we could imagine having a way to group them by prefix.

Reply:
AFAIR we said that it should be the case for `<default>`, because it reflects that fact that we know the destination, which in RabbitMQ is named with an empty string. This will distinguish such from others where the destination is really unknown. 
The thing with polling spans is that they are expected to result with unknown destination in some frameworks when they end without a message/record.

Reply:
I had an offline discussion with @SylvainJuge about it and he raised a good point that adding `<unknown>` can help with setting the `ELASTIC_APM_IGNORE_MESSAGE_QUEUES` for removing polling spans that end without a known destination. We will leave it as is and see if we get any feedback on that. For now, I will only remove the `rabbitmq` example.